x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -51,3 +68,65 @@ export const transformValidateBulkError = (
     });
   }
 };
+
+// Temporary functionality until the new System Actions is in place
+// for now we want to make sure that user cannot configure Isolate action if has no RBAC permissions to do so
+export const validateResponseActionsPermissions = async (
+  securitySolution: SecuritySolutionApiRequestHandlerContext,
+  ruleUpdate: RuleCreateProps | RuleUpdateProps,
+  existingRule?: RuleAlertType | null
+): Promise<void> => {
+  const { experimentalFeatures } = await securitySolution.getConfig();
+
+  if (!experimentalFeatures.endpointResponseActionsEnabled || !isQueryRule(ruleUpdate.type)) {
+    return;
+  }
+
+  const payload = ruleUpdate as QueryRule;
```

Comment:
Type casting `as QueryRule` and `Rule<UnifiedQueryRuleParams` isn't necessary anymore as type guards are used.